### PR TITLE
Twig compatibility (v1, v2 & v3)

### PR DESCRIPTION
**Pull request by atomiix**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Even though we updated the range of twig compatibility in the composer.json, this lib was not really compatible with twig > 1. This PR aims to make it compatible with twig from v1 to v3.
| Type?             | improvement
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | 
| How to test?      | CI should be green
| Possible impacts? | 

### BC Break
The minimum version of PHP is now 7.2

### CODE REVIEW

File: Resources/config/services.yml [hunk 1]
```diff
@@ -65,4 +65,4 @@ services:
           - { name: prestashop.translation.extractor, format: smarty }
 
     prestashop.dumper.xliff:
-        class: PrestaShop\TranslationToolsBundle\Translation\Dumper\XliffFileDumper
+        class: PrestaShop\TranslationToolsBundle\Translation\Dumper\XliffFileDumper
```
Comment (matks) [inline Resources/config/services.yml:68]: Missing empty line

File: Tests/Translation/Extractor/TwigExtractorTest.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
```
Comment (matks) [inline Tests/Translation/Extractor/TwigExtractorTest.php:7]: This repository is supposed to be MIT https://github.com/PrestaShop/TranslationToolsBundle/blob/master/composer.json#L5

File: Twig/Lexer.php [hunk 3]
```diff
@@ -2,39 +2,36 @@
...
+    public function tokenize($code, $name = null): TokenStream
     {
-        parent::lexComment();
-
-        if (true === $this->call) {
-            return;
+        if (!$code instanceof Source) {
+            $source = new Source($code, $name);
+        } else {
+            $source = $code;
```
Comment (matks) [inline Twig/Lexer.php:20]:
```
$source = $code;
$code = $source->getCode();
```
So somehow `$code = $code->getCode();` 😝  ?

**Review by matks — COMMENTED** [review]
Logic seems OK. 2 very small issues
- one missing empty line
- one license header wrong

If you fix them, I approve the PR 😉 

Nice work on the CI !

File: DependencyInjection/TranslationToolsExtension.php [hunk 4]
```diff
@@ -36,6 +36,13 @@ class TranslationToolsExtension extends Extension
 {
     public function load(array $configs, ContainerBuilder $container)
     {
+        $definition = $container->register(
+            'translation_tools.translation.extension',
+            'PrestaShop\TranslationToolsBundle\Twig\Extension\TranslationExtension'
```
Comment (PierreRambaud) [inline DependencyInjection/TranslationToolsExtension.php:41]:
```suggestion
            TranslationExtension::class
```
?

File: Twig/Lexer.php (hunk 3 above)
Comment (atomiix) [reply Twig/Lexer.php:20]: Yes, I've done it on purpose, I know it's a bit weird but that's kinda how twig itself handle it to be retrocompatible (https://github.com/twigphp/Twig/blob/1.x/src/Lexer.php#L165-L183)

File: Translation/Extractor/TwigExtractor.php [hunk 5]
```diff
@@ -63,12 +63,18 @@ class TwigExtractor extends BaseTwigExtractor implements ExtractorInterface
...
+    public function __construct(Environment $twig)
     {
         $this->twig = $twig;
         $this->twigLexer = new Lexer($this->twig);
+
+        $this->twig->registerUndefinedFunctionCallback(function () {
+        });
+
+        $this->twig->registerUndefinedFilterCallback(function () {
+        });
```
Comment (PierreRambaud) [inline Translation/Extractor/TwigExtractor.php:77]:
```suggestion
        $this->twig->registerUndefinedFunctionCallback(function () {});

        $this->twig->registerUndefinedFilterCallback(function () {});
```

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thank you @atomiix

**Pull request merged**